masonry/src/render_root.rs

```diff
@@ -497,44 +453,22 @@ impl RenderRoot {
         kurbo::Size::new(size.width, size.height)
     }
 
-    // --- MARK: POST-EVENT ---
-    fn post_event_processing(&mut self, widget_state: &mut WidgetState) {
-        // If children are changed during the handling of an event,
-        // we need to send RouteWidgetAdded now, so that they are ready for update/layout.
-        if widget_state.children_changed {
-            run_update_new_widgets_pass(self);
-        }
-
-        if self.state.debug_logger.layout_tree.root.is_none() {
-            self.state.debug_logger.layout_tree.root = Some(self.root.id().to_raw() as u32);
-        }
-
-        if !self.state.scroll_request_targets.is_empty() {
-            run_update_scroll_pass(self);
-        }
-
-        if self.root_state().needs_compose && !self.root_state().needs_layout {
-            root_compose(self, widget_state);
-        }
-
-        // Update the disabled and stashed state if necessary
-        // Always do this before updating the focus-chain
-        if self.root_state().needs_update_disabled {
-            run_update_disabled_pass(self);
-        }
-        if self.root_state().needs_update_stashed {
-            run_update_stashed_pass(self);
-        }
-
-        // Update the focus-chain if necessary
-        // Always do this before sending focus change, since this event updates the focus chain.
-        if self.root_state().update_focus_chain {
-            run_update_focus_chain_pass(self);
-        }
-
-        run_update_focus_pass(self, widget_state);
-
-        if self.root_state().request_anim {
+    // --- MARK: REWRITE PASSES ---
+    pub(crate) fn run_rewrite_passes(&mut self) {
+        // TODO - Rerun passes if invalidation flags are still set
+
+        run_mutate_pass(self);
+        run_update_widget_tree_pass(self);
+        run_update_disabled_pass(self);
+        run_update_stashed_pass(self);
+        run_update_focus_chain_pass(self);
+        run_update_focus_pass(self);
+        root_layout(self);
+        run_update_scroll_pass(self);
+        root_compose(self);
+        run_update_pointer_pass(self);
```

Comment:
Why not the method syntax here? (in general for passes and all the functions there, seems (almost) every function in the passes has `&mut RenderRoot` as parameter)

Reply:
I'm a little hesitant to add impl blocks for RenderRoot outside of `render_root.rs`. Doing so seems just a little too surprising.

Calling free functions with self is a little surprising, but it's not actually more verbose, so I'd say it's fine.